### PR TITLE
Psi ping fix

### PR DESCRIPTION
**Pull request by Snakebittenn**
Holy fuck.
Anyway, it now properly checks for psiaugs.

### CODE REVIEW

**Comment by Snakebittenn** [comment]
!bugfix

**Comment by Snakebittenn** [comment]
!review

File: code/modules/psionics/faculties/coercion.dm [hunk 1]
```diff
@@ -300,7 +300,7 @@
 			if(!T || L == user || L.stat == DEAD || L.invisibility == INVISIBILITY_LEVEL_TWO)
 				continue
 			if(L.isSynthetic() || L.is_diona() || isvaurca(L))
-				var/obj/item/organ/internal/augment/psiaug = locate() in L
+				var/obj/item/organ/internal/augment/psi/psiaug = locate() in L
```
Comment (mikomyazaki) [inline code/modules/psionics/faculties/coercion.dm:303]:
```suggestion
				psiaug = has_psi_aug()
```

Or possibly use is_psi_blocked()

Comment (Snakebittenn) [reply code/modules/psionics/faculties/coercion.dm]: travis shit the bed on that

File: code/modules/psionics/faculties/coercion.dm [hunk 2]
```diff
@@ -300,7 +300,7 @@
 			if(!T || L == user || L.stat == DEAD || L.invisibility == INVISIBILITY_LEVEL_TWO)
 				continue
 			if(L.isSynthetic() || L.is_diona() || isvaurca(L))
-				var/obj/item/organ/internal/augment/psiaug = locate() in L
+				psiaug = has_psi_aug()
```
Comment (NonQueueingMatt) [inline code/modules/psionics/faculties/coercion.dm:303]: You... forgot... to specify what actually the proc should be called on.

**Comment by Snakebittenn** [comment]
i'm just going to go back to how it was

File: code/modules/psionics/faculties/coercion.dm (hunk 1 above)
Comment (mikomyazaki) [reply code/modules/psionics/faculties/coercion.dm]: Yes, you just need to call the proc on the mob instead.

**Comment by skull132** [comment]
uuuuh. _wat_.

File: code/modules/psionics/faculties/coercion.dm [hunk 3]
```diff
@@ -300,7 +300,7 @@
 			if(!T || L == user || L.stat == DEAD || L.invisibility == INVISIBILITY_LEVEL_TWO)
 				continue
 			if(L.isSynthetic() || L.is_diona() || isvaurca(L))
-				var/obj/item/organ/internal/augment/psiaug = locate() in L
+				var/obj/item/organ/internal/augment/psiaug/psiaug = locate() in L
```
Comment (NonQueueingMatt) [inline code/modules/psionics/faculties/coercion.dm:303]: this is the part where i question what the hell is going on with this
I'm looking at the `has_psi_aug` proc and it returns the following item: `/obj/item/organ/internal/augment/psi`
But here you're checking for `psiaug`?
Also, why aren't you using the proc you made for this?

File: code/modules/psionics/faculties/coercion.dm [hunk 4]
```diff
@@ -300,7 +300,7 @@
 			if(!T || L == user || L.stat == DEAD || L.invisibility == INVISIBILITY_LEVEL_TWO)
 				continue
 			if(L.isSynthetic() || L.is_diona() || isvaurca(L))
```
Comment (MarinaGryphon) [inline code/modules/psionics/faculties/coercion.dm:302]:
```suggestion
			var/psi_is_blocked = L.is_psi_blocked()
			if(psi_is_blocked)
						continue
```
then delete everything up to the next `continue`

File: code/modules/psionics/faculties/coercion.dm (hunk 3 above)
Comment (MarinaGryphon) [inline code/modules/psionics/faculties/coercion.dm:303]:
```suggestion
```

File: code/modules/psionics/faculties/coercion.dm [hunk 5]
```diff
@@ -300,7 +300,7 @@
...
-				var/obj/item/organ/internal/augment/psiaug = locate() in L
+				var/obj/item/organ/internal/augment/psiaug/psiaug = locate() in L
 				if(!psiaug)
```
Comment (MarinaGryphon) [inline code/modules/psionics/faculties/coercion.dm:304]:
```suggestion
```

File: code/modules/psionics/faculties/coercion.dm [hunk 6]
```diff
@@ -300,7 +300,7 @@
...
 					continue
```
Comment (MarinaGryphon) [inline code/modules/psionics/faculties/coercion.dm:305]:
```suggestion
```

File: code/modules/psionics/faculties/coercion.dm (hunk 1 above)
Comment (MarinaGryphon) [reply code/modules/psionics/faculties/coercion.dm]: has_psi_aug() returns boolean anyway

File: code/modules/psionics/faculties/coercion.dm (hunk 2 above)
Comment (MarinaGryphon) [reply code/modules/psionics/faculties/coercion.dm]: also it returns boolean

File: code/modules/psionics/faculties/coercion.dm [hunk 7]
```diff
@@ -299,10 +299,9 @@
 			var/turf/T = get_turf(L)
 			if(!T || L == user || L.stat == DEAD || L.invisibility == INVISIBILITY_LEVEL_TWO)
 				continue
-			if(L.isSynthetic() || L.is_diona() || isvaurca(L))
-				var/obj/item/organ/internal/augment/psiaug/psiaug = locate() in L
-				if(!psiaug)
-					continue
+			var/psi_is_blocked = L.is_psi_blocked()
+			if(psi_is_blocked)
+				continue
```
Comment (mikomyazaki) [inline code/modules/psionics/faculties/coercion.dm:304]:
```suggestion
			if(L.is_psi_blocked())
				continue
```

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

**Review by mikomyazaki — APPROVED** [review]

**Review by Geevies — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**